### PR TITLE
Optimize iRODS ``_exists()`` method

### PR DESCRIPTION
**Pull request by kxk302**
To improve the performance of irods calls

### CODE REVIEW

**Review by nsoranzo — COMMENTED** [review]
If that's correct, then you need to remove the `in_cache` variable from this method.

**Comment by kxk302** [comment]
> If that's correct, then you need to remove the `in_cache` variable from this method.

That is correct. Please see the revised code I just pushed. Thx

File: lib/galaxy/objectstore/irods.py [hunk 1]
```diff
@@ -454,20 +454,20 @@ def file_ready(self, obj, **kwargs):
         return False
 
     def _exists(self, obj, **kwargs):
-        in_cache = in_irods = False
+        in_irods = False
```
Comment (mvdbeek) [inline lib/galaxy/objectstore/irods.py]: You can simplify this method further. The github interface doesn't let me create a nice multi-line suggestion here, but this is what it boils down to:
```python
if self._in_cache(rel_path) or self._data_object_exists(rel_path):
    return True
```
At this point the only way we can still return `True` is if `dir_only ` and `base_dir` are both not "False-y".
So you can do
```python
if dir_only and base_dir:
    if not os.path.exists(rel_path):
        os.makedirs(rel_path)
    return True
return False
```

Comment (kxk302) [reply lib/galaxy/objectstore/irods.py]: Good idea. Revised code accordingly.

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**